### DISCUSSION TITLE
Mount credentials into local execution docker container

### DISCUSSION THREAD

**Discussion by joekitsmith** [kubeflow/pipelines#10822] [General]
I'm using kfp to run a pipeline in GCP Vertex AI. I have an e2e test setup that will deploy the pipeline and check the outputs but this takes a long time to run. I want to be able to run my pipeline locally to check everything is configured correctly.

I have been trying to use [kfp local execution](https://www.kubeflow.org/docs/components/pipelines/v2/local-execution/) with `DockerRunner`. I am using container components so this would work nicely.

However, my component requires GCP credentials to access secrets that are used to retrieve data from a third party application. These credentials are not present in the local docker container that is created and so retrieval of the secrets fails.

Is there any way I can mount my local GCP credentials into the docker container via kfp?

**Comment by carlthome** [comment]
Also wondering this!

I'm trying to develop a Google Cloud Secret Manager component and want to mount my developer credentials into the local container.

**Comment by gleb-lukicov** [comment]
+1 I would also like to know a solution for this!

**Comment by dshkliarenko** [comment]
A workaround which works for me is to monkey-patch `DockerTaskHandler.get_volumes_to_mount` adding mount mapping for credentials path, and then set env variable pointing to that path inside the component which requires the credentials.

**Comment by EladProject** [comment]
I'm using pipeline_root to pass a .env file to the component (a terrible solution, I know). Maybe you can use it to pass the credentials json.

**Comment by DevoChristian** [comment]
It's not pretty, but kfp local execution automatically mounts the output directory local_outputs/. You can place your ADC file here and use it for local auth at runtime. This works, but again, is not pretty.
